src/content_cao.cpp

```diff
@@ -1302,14 +1308,17 @@ void GenericCAO::updateTexturePos()
 	}
 }
 
-void GenericCAO::updateTextures(const std::string &mod)
```

Comment:
I think this is when m_previous_texture_modifier is passed in.

On enter of updateTextures:

```
mod = m_previous_texture_modifier = "foo"
m_previous_texture_modifier = "foo"
m_current_texture_modifier = "bar"
```

after m_previous_texture_modifier = m_current_texture_modifier

```
mod = m_previous_texture_modifier = "bar"
m_previous_texture_modifier = "bar"
m_current_texture_modifier = "bar"
```

after m_current_texture_modifier = mod

```
mod = m_previous_texture_modifier = "bar"
m_previous_texture_modifier = "bar"
m_current_texture_modifier = "bar"
```

So even though mod is const, it isn't copied and still points to m_previous_texture_modifier which causes problems when it changes